src/backend/tools/web_scrape.py

```diff
@@ -0,0 +1,33 @@
+from typing import Any, Dict, List
+
+from bs4 import BeautifulSoup
+from requests import get
+
+from backend.tools.base import BaseTool
+
+
+class WebScrapeTool(BaseTool):
+    NAME = "web_scrape"
+
+    @classmethod
+    def is_available(cls) -> bool:
+        return True
+
+    async def call(self, parameters: dict, **kwargs: Any) -> List[Dict[str, Any]]:
+        url = parameters.get("url")
+
+        response = get(url)
+        if not response.ok:
```

Comment:
Wondering if we should return an exception here or a dict containing the error. The python interpreter, for example, returns the std_err when the code fails to execute, so the model knows the issue and tries to correct it. I'm not sure if this could lead to infinite loops here, but it might be worth trying

Reply:
hmm thats a great point. lets return dict with error in `text` and see how that works out.